pulser/simulation/simresults.py

```diff
@@ -14,63 +14,336 @@
+    """Results of a noisy simulation run of a pulse sequence.
+
+    Contrary to a CleanResults object, this object contains a list of Counter
+    describing the state distribution at the time it was created by using
+    Simulation.run() with a noisy simulation.
+    Contains methods for studying the populations and extracting useful
+    information from them.
+    """
+
+    def __init__(self, run_output: list[Counter],
+                 size: int, basis_name: str,
+                 sim_times: np.ndarray, n_measures: int, dim: int = 2) -> None:
+        """Initializes a new NoisyResults instance.
+
+        Warning:
+            Can't have single-atom Hilbert spaces with dimension bigger
+            than 2 for NoisyResults objects.
+            This is not the case for a CleanResults object, containing states
+            in Hilbert space, but NoisyResults contains a probability
+            distribution of bitstrings, not atomic states
+
+        Args:
+            run_output (list[Counter]): Each Counter contains the
+                probability distribution of a multi-qubits state,
+                represented as a bitstring. There is one Counter for each time
+                the simulation was asked to return a result.
+            size (int): The number of atoms in the register.
+            basis_name (str): Basis indicating the addressed atoms after
+                the pulse sequence ('ground-rydberg' or 'digital' - 'all' basis
+                makes no sense after projection on bitstrings).
+            sim_times (list): Times at which Simulation object returned the
+                results.
+            meas_basis (Optional[str]): The basis in which a sampling
+                measurement is desired.
+            n_measures (int): Number of measurements needed to compute this
+                result when doing the simulation.
+            dim (int): Equal to 2 here, since projections already happened.
+        """
+        if basis_name == 'all':
+            raise ValueError("`basis_name` must be either 'ground-rydberg' or"
+                             + " 'digital'.")
+        super().__init__(dim, size, basis_name, sim_times)
+        self.n_measures = n_measures
+        self._results = run_output
+
+    @property
+    def states(self) -> list[qutip.Qobj]:
+        """Measured states as a list of diagonal qutip.Qobj."""
+        return [self.get_state(t) for t in self._sim_times]
+
+    @property
+    def results(self) -> list[Counter]:
+        """Probability distribution of the bitstrings."""
+        return self._results
+
+    def get_state(self, t: float, t_tol: float = 1.e-3) -> qutip.Qobj:
+        """Get the state at time t as a diagonal density matrix.
+
+        Note:
+            This is not the density matrix of the system, but is a convenient
+            way of computing expectation values of observables.
+
+        Args:
+            t (int): Time of the state to be returned.
```

Comment:
Specify the units, change to float and perhaps use the same message as in `CleanResults` (or vice-versa)